src/com/amazon/kinesis/streaming/agent/tailing/SourceFile.java

```diff
@@ -114,6 +115,13 @@ public String toString() {
     private void validateDirectory(Path dir) {
         Preconditions.checkArgument(dir != null, "Directory component is empty!");
         // TODO: validate that the directory component has no glob characters
+
+        //check the permissions of every level in the directory path
+        for(Path level : dir.toAbsolutePath()){
```

Comment:
My understanding is that .toAbsolutePath() will return a path object. 

Since the path class extends iterable, this loop should allow us to check each directory individually as per this documentation (See the very end: https://docs.oracle.com/javase/tutorial/essential/io/pathOps.html). Is this correct?